helm-chart/ray-cluster/templates/raycluster-cluster.yaml

```diff
@@ -14,6 +14,17 @@ spec:
   {{- if .Values.head.enableInTreeAutoscaling }}
   enableInTreeAutoscaling: {{ .Values.head.enableInTreeAutoscaling }}
   {{- end }}
+  {{- if .Values.head.autoscalerOptions }}
+  {{- $values := .Values.head.autoscalerOptions}}
+  autoscalerOptions: 
+    upscalingMode: {{ $values.upscalingMode | default "Default" }}
+    idleTimeoutSeconds: {{ $values.idleTimeoutSeconds | default 60 }}
+    image: {{ $values.image | default (printf "%s:%s" .Values.image.repository .Values.image.tag) }}
+    imagePullPolicy: {{ $values.imagePullPolicy | default "Always" }}
+    {{- if $values.resources }}
+    resources: {{- toYaml $values.resources | nindent 6 }}
+    {{- end }}
+  {{- end }}
```

Comment:
In addition, it is unnecessary to expand all fields of `autoscalerOptions` in `raycluster-cluster.yaml`. You can use `toYaml` ([example](https://github.com/ray-project/kuberay/blob/master/helm-chart/ray-cluster/templates/raycluster-cluster.yaml#L36)) to pass it from `values.yaml` to `raycluster-cluster.yaml` in one line.

```
autoscalerOptions: {{- toYaml .Values.head.autoscalerOptions | nindent 4}}
```

Reply:
+1